### PR TITLE
Use o2mIndex for matset converters + refactor

### LINKED ISSUE #1238: Use O2M index for matset converters
https://github.com/LLNL/conduit/blob/develop/src/libs/blueprint/conduit_blueprint_o2mrelation_index.hpp

### PR DESCRIPTION
**Pull request by JustinPrivitera**
Resolves #1238

We can get this in to 0.9.1 if you'd like, or just punt on it until later. I think there is more I could add here that would clean things up.

### CODE REVIEW

File: src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp [hunk 1]
```diff
@@ -503,6 +504,167 @@ to_silo(const conduit::Node &field,
...
+                      std::map<int, std::string> &reverse_matmap)
+{
+    // fill out map
+    auto matmap_itr = src_matset["material_map"].children();
+    while (matmap_itr.has_next())
+    {
+        const Node &matmap_entry = matmap_itr.next();
+        const std::string matname = matmap_itr.name();
+
+        reverse_matmap[matmap_entry.as_int()] = matname;
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp]: to_int64 (as_int could fail)

Comment (cyrush) [reply src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp]: or `to_int`

**Review by cyrush — COMMENTED** [review]
Looks good -- one more small type related suggestion

File: src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp (hunk 1 above)
Comment (JustinPrivitera) [reply src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp]: ok done

**Review by cyrush — APPROVED** [review]

**Pull request merged**